src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -244,21 +249,19 @@ public OpenDistroSecurityPlugin(final Settings settings, final Path configPath)
         sslCertReloadEnabled = isSslCertReloadEnabled(settings);
 
         if (disabled) {
-            this.dlsFlsAvailable = false;
             this.advancedModulesEnabled = false;
             this.sslCertReloadEnabled = false;
             log.warn("Open Distro Security plugin installed but disabled. This can expose your configuration (including passwords) to the public.");
             return;
         }
-        
+
         if (openDistroSSLConfig.isSslOnlyMode()) {
-            this.dlsFlsAvailable = false;
             this.advancedModulesEnabled = false;
```

Comment:
Do we still need advancedModulesEnabled flag? 

if it is not SslOnlyMode, means all features are enabled.

Reply:
Yes we can. `advancedModulesEnabled` is equal to not ssl-only and not-disabled. 
Since we are reading off settings, I had deprecated the setting in this PR. Can remove the whole flag if needed.

What are your thoughts ?

Reply:
Yes, I think we should remove the flag.. as it is not in effect as such based on current code logic.. we can keep warning as this flag was never in effect logically..